hudi-client/src/test/java/org/apache/hudi/TestHoodieClientOnCopyOnWriteStorage.java

```diff
@@ -515,6 +516,161 @@ public void testSmallInsertHandlingForInserts() throws Exception {
         inserts1.size() + inserts2.size() + insert3.size());
   }
 
+  /**
```

Comment:
@nsivabalan :  Can you see if you add additional test-block (for testing deletes) to existing test like testUpsertsInternal ? Keeps test code manageable, avoids duplicate test-setups and also can be used to test diffferent combination (upsert/upsertPrepped followed by delete) ?